test/integration/associations/scope.test.js

```diff
@@ -321,96 +315,95 @@ describe(Support.getTestDialectTeaser('associations'), () => {
-                  { model: this.Tag, as: 'categories' }
-                ]
-              }), this.Post.findOne({
-                where: {
-                  id: this.postC.get('id')
-                },
-                include: [
-                  { model: this.Tag, as: 'tags' },
-                  { model: this.Tag, as: 'categories' }
-                ]
-              })]);
-            }).then(([postA, postB, postC]) => {
-              expect(postA.get('categories').length).to.equal(1);
-              expect(postA.get('tags').length).to.equal(1);
-              expect(postB.get('categories').length).to.equal(1);
-              expect(postB.get('tags').length).to.equal(1);
-              expect(postC.get('categories').length).to.equal(1);
-              expect(postC.get('tags').length).to.equal(1);
-
-              expect(postA.get('categories')[0].get('type')).to.equal('category');
-              expect(postA.get('tags')[0].get('type')).to.equal('tag');
-              expect(postB.get('categories')[0].get('type')).to.equal('category');
-              expect(postB.get('tags')[0].get('type')).to.equal('tag');
-              expect(postC.get('categories')[0].get('type')).to.equal('category');
-              expect(postC.get('tags')[0].get('type')).to.equal('tag');
-            });
+          it('should create, find and include associations with scope values', async function() {
+            await Promise.all([this.Post.sync({ force: true }), this.Tag.sync({ force: true })]);
+            await this.PostTag.sync({ force: true });
+
+            const [postA0, postB0, postC0, categoryA, categoryB, tagA, tagB] = await Promise.all([
+              this.Post.create(),
+              this.Post.create(),
+              this.Post.create(),
+              this.Tag.create({ type: 'category' }),
+              this.Tag.create({ type: 'category' }),
+              this.Tag.create({ type: 'tag' }),
+              this.Tag.create({ type: 'tag' })
+            ]);
+
+            this.postA = postA0;
+            this.postB = postB0;
+            this.postC = postC0;
+
+            await Promise.all([
+              postA0.addCategory(categoryA),
+              postB0.setCategories([categoryB]),
+              postC0.createCategory(),
+              postA0.createTag(),
+              postB0.addTag(tagA),
+              postC0.setTags([tagB])
+            ]);
+
+            const [postACategories, postATags, postBCategories, postBTags, postCCategories, postCTags] = await Promise.all([
+              this.postA.getCategories(),
+              this.postA.getTags(),
+              this.postB.getCategories(),
+              this.postB.getTags(),
+              this.postC.getCategories(),
+              this.postC.getTags()
+            ]);
+
+            expect(postACategories.length).to.equal(1);
+            expect(postATags.length).to.equal(1);
+            expect(postBCategories.length).to.equal(1);
+            expect(postBTags.length).to.equal(1);
```

Comment:
@sushantdhiman this assertion failed in an unrelated PR: https://ci.appveyor.com/project/sushantdhiman/sequelize/builds/33045550

But I can't see anything wrong with the code changes to this test...hopefully there's not anything I broke in core?  Have you been seeing failures here periodically?

Reply:
https://github.com/sequelize/sequelize/blob/e34efd477d051af6796c277406be26bb1d2c2f23/test/integration/associations/scope.test.js#L318

I have noticed this test failing a few times now as well. AFAIR only fails for MSSQL, will try to figure out what is going on if I get chance. Perhaps some promise is not returned somewhere in core codebase

Reply:
Do you think it started failing after I asyncified core?

Reply:
Can't say exactly when it started failing but looks related to `bluebird removal` / `async conversion`

Reply:
Didn't we have another case where the bluebird removal made code fast enough to uncover an unchained promise/race condition?

Reply:
Heh, I'm kinda confused, didn't Bluebird bill itself as being the fastest promise implementation out there?  Or did that change once promises became native in V8?

Reply:
Ah yup it changed by Node 12: https://blog.kuzzle.io/bluebird-vs-native-vs-async/await-state-of-promises-performances-in-2019
https://v8.dev/blog/fast-async